### PR TITLE
Make model selector metadata to metric more robust

### PR DESCRIPTION
**Pull request by erica-chiu**
**Related issues**

**Describe the proposed solution**
Change the matching when parsing model selector metadata into metrics from name of metric to metric keys

**Describe alternatives you've considered**
N/A

**Additional context**
Allows for parsing of metadata to be based on content rather than the name, allowing for multiple metrics of the same type

### CODE REVIEW

File: core/src/main/scala/com/salesforce/op/stages/impl/selector/ModelSelectorSummary.scala [hunk 1]
```diff
@@ -240,19 +240,26 @@ case object ModelSelectorSummary {
...
+                case `binary` =>
                   nm -> JsonUtils.fromString[BinaryClassificationMetrics](valsJson).get
-                case OpEvaluatorNames.BinScore.humanFriendlyName =>
+                case `binscore` =>
                   nm -> JsonUtils.fromString[BinaryClassificationBinMetrics](valsJson).get
-                case OpEvaluatorNames.Multi.humanFriendlyName =>
+                case `multi` =>
                   nm -> JsonUtils.fromString[MultiClassificationMetrics](valsJson).get
-                case OpEvaluatorNames.Regression.humanFriendlyName =>
+                case `regression` =>
```
Comment (leahmcguire) [inline core/src/main/scala/com/salesforce/op/stages/impl/selector/ModelSelectorSummary.scala]: hmm, it is better not to have this hard coded as text, but I noticed a problem with matching in Enumns in my PR as well... @tovbinm is this related to the upgrade?

Comment (tovbinm) [reply core/src/main/scala/com/salesforce/op/stages/impl/selector/ModelSelectorSummary.scala]: Weird. Looking at it...

**Comment by tovbinm** [comment]
@erica-chiu proposed tests are defining the custom evaluator incorrectly. Here is how is supposed to work #387
